### PR TITLE
Custom help content for "Jetpack required Error" screen

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #7547 

### Description
As a part of improving the help center content, this PR adds custom content with FAQs and Answers for the "Jetpack required Error" screen at the following URL.

https://woocommerce.com/document/android-ios-apps-login-help-faq/#jetpack-required

Internal ref - pe5sF9-sZ-p2

### Testing instructions

**You will need a store with no Jetpack plugin.** 

1. Install and launch the app.
1. Tap `Skip` to reach the login prologue screen.
1. Tap the `Enter Your Store Address` button.
1. Enter your store address (with no Jetpack plugin installed) and login using your WordPress.com
1. You will land in the "Jetpack required error" screen from which you can choose to install Jetpack or login using different account.
1. Tap `Help` -> `Help Center`
1. Observe that you are directed to https://woocommerce.com/document/android-ios-apps-login-help-faq/#jetpack-required in a web view
1. In Xcode debug console observe that the following event is tracked. Note that the `source_flow` property will differ based on the login method you selected. 
```
Tracked support_help_center_viewed, properties: [AnyHashable("help_content_url"): "https://woocommerce.com/document/android-ios-apps-login-help-faq/#jetpack-required", AnyHashable("source_flow"): "login_password_magic_link_emphasis", AnyHashable("source_step"): "jetpack_not_connected"]
```

### Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift [hunk 1]
```diff
@@ -379,6 +379,18 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
         }
     }
 
+    /// Presents the Support Interface
+    ///
+    /// - Parameters:
+    ///     - from: UIViewController instance from which to present the support interface
+    ///     - screen: A case from `CustomHelpCenterContent.Screen` enum. This represents authentication related screens from WCiOS.
+    ///
+    func presentSupport(from sourceViewController: UIViewController, screen: CustomHelpCenterContent.Screen) {
```
Comment (jaclync) [inline WooCommerce/Classes/Authentication/AuthenticationManager.swift:388]: nit: now that there are multiple variations of calls to `presentSupport(from:customHelpCenterContent:)`, how about making `presentSupport(from:customHelpCenterContent:)` the function in the protocol and moving the ones that call this function to the protocol's extension? if there's another variation in the future, we don't have to change the interface and every implementation every time.

something like:

```swift
protocol Authentication {
    func presentSupport(from sourceViewController: UIViewController,
                                        customHelpCenterContent: CustomHelpCenterContent?)
    ...
}

extension Authentication {
    func presentSupport(from sourceViewController: UIViewController, screen: CustomHelpCenterContent.Screen) {
        let customHelpCenterContent = ...
        presentSupport(from: sourceViewController, customHelpCenterContent: customHelpCenterContent) 
    }

    func presentSupport(from sourceViewController: UIViewController,
                        sourceTag: WordPressSupportSourceTag,
                        lastStep: AuthenticatorAnalyticsTracker.Step,
                        lastFlow: AuthenticatorAnalyticsTracker.Flow) {
        let customHelpCenterContent = ...
        presentSupport(from: sourceViewController, customHelpCenterContent: customHelpCenterContent) 
    }
}

File: WooCommerce/WooCommerceTests/Authentication/ULErrorViewControllerTests.swift [hunk 2]
```diff
@@ -81,6 +81,50 @@ final class ULErrorViewControllerTests: XCTestCase {
         XCTAssertEqual(auxiliaryButtonHidden, viewModel.isAuxiliaryButtonHidden)
     }
 
+    func test_viewcontroller_does_not_have_right_bar_button_item_when_rightBarButtonItemTitle_is_nil_in_viewmodel() throws {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Authentication/ULErrorViewControllerTests.swift:84]: nice tests!

**Review by jaclync — APPROVED** [review]
Works as expected! 🚢

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/Authentication/AuthenticationManager.swift:388]: I don't fully understand this. But, I will share some more details down below. 

1. The method to present support using sourceTag, step and flow is from [the WordPressAuthenticator library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift#L71).  So, I don't think adding that to an extension to `Authentication` protocol will make sense.
2. I plan to remove [the present using sourceTag method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L19) from [Authentication protocol](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift) and replace those calls with [custom help center method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L15). I will delete [this method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L19) once I implement custom help centers for `RoleErrorViewController` and `StorePickerViewController`. i.e. Using `sourceTag` inside WCiOS screens will be removed, as the passed in parameter `sourceTag` is not used anywhere.

Kindly let me know if you think this sounds good enough.

Comment (jaclync) [reply WooCommerce/Classes/Authentication/AuthenticationManager.swift:388]: > 1. The method to present support using sourceTag, step and flow is from [the WordPressAuthenticator library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift#L71).  So, I don't think adding that to an extension to `Authentication` protocol will make sense.

ohh I wasn't aware that `presentSupport(from:sourceTag:lastStep:lastFlow:)` is from `WordPressAuthenticatorDelegate` and not `Authentication` protocol 😅 Please ignore the suggestion then.

The plan for the future changes sounds good, feel free to merge the PR then 🚀

**Pull request merged**